src/components/Register.jsx

```diff
@@ -7,26 +7,45 @@ import {
   chakra,
   Text,
   InputLeftElement,
-  InputGroup,
+  InputGroup
 } from '@chakra-ui/react';
-import { Link } from 'react-router-dom';
+import { Link, useNavigate } from 'react-router-dom';
 import { FiUser, FiLock } from 'react-icons/fi';
 import { HiOutlineMail } from 'react-icons/hi';
 import { useForm } from 'react-hook-form';
+import axios from 'axios';
 
 const CFiUser = chakra(FiUser);
 const CHiOutlineMail = chakra(HiOutlineMail);
 const CFiLock = chakra(FiLock);
 
-const Register = () => {
+const Register = (props) => {
+  const navigate = useNavigate();
   const { register, handleSubmit } = useForm();
-  const onSubmit = (d) => {
-    console.log("test register", JSON.stringify(d));
-    alert('Uspesno ste registrovani!');
+  const onSubmit = (user) => {
+    console.log('test register', JSON.stringify(user));
+    axios
+      .post('http://localhost:1337/api/auth/local/register', {
+        username: user.name,
+        email: user.email,
+        password: user.password
+      })
+      .then((response) => {
+        // Handle success.
+        console.log('Well done!');
+        console.log('User profile', response.data.user);
+        console.log('User token', response.data.jwt);
+        alert('Uspesno ste registrovani!');
+        navigate('/');
+      })
+      .catch((error) => {
+        // Handle error.
+        console.log('An error occurred:', error.response);
+      });
```

Comment:
- Please let's move this ajax call to `services/auth.js`
- Also let's move the login call from auth context to `services/auth.js`
- `services/auth.js` should use `services/http.js` - that service should also be created, and its purpose should be to create an axios instance, set some default values (like `baseURL`) and export it. Then you should use `services/http.js` instead of axios directly whenever you need to make an ajax call
- The flow should be like this: 
  - Register component calls a function to handle registration from AuthContext, and if successful, it redirects to '/'
  - The function from auth context calls the `register` function from `services/auth.js`
  - The `register` function from `services/auth.js` does the ajax call using the `services/http.js`

The reason for all of this is to keep the code modular, and to separate concerns (https://deviq.com/principles/separation-of-concerns) - in this case for example, we don't want to have one function that handles form UI, calling the API and browser navigation, because those are all different concerns

Also, after you refactor this, we should store the JWT and user data in the AuthContext and localStorage.

Additionally, please checkout JIRA task TEAM-33 for instructions on how to create all required register functionality